### PR TITLE
feat(metadata): Allow querying by metadata.

### PR DESCRIPTION
**Pull request by christiandompierre**
Relates to https://github.com/uc-cdis/indexclient/pull/17

### CODE REVIEW

File: indexd/alias/blueprint.py [hunk 1]
```diff
@@ -59,7 +59,7 @@ def get_alias():
 
     start = flask.request.args.get('start')
 
-    hashes = flask.request.args.getlist('hash')
+    hashes = flask.request.args.getlist('hashes')
```
Comment (philloooo) [inline indexd/alias/blueprint.py]: should support both for backward compatibility

File: indexd/index/blueprint.py [hunk 2]
```diff
@@ -73,12 +73,15 @@ def get_index():
 
     version = flask.request.args.get('version')
 
-    hashes = flask.request.args.getlist('hash')
+    hashes = flask.request.args.getlist('hashes')
```
Comment (philloooo) [inline indexd/index/blueprint.py]: same as previous comment

File: indexd/index/blueprint.py [hunk 3]
```diff
@@ -73,12 +73,15 @@ def get_index():
...
     hashes = {h: v for h, v in map(lambda x: x.split(':', 1), hashes)}
 
     validate_hashes(**hashes)
     hashes = hashes if hashes else None
 
+    metadata = flask.request.args.getlist('metadata')
+    metadata = {k: v for k, v in map(lambda x: x.split(':', 1), metadata)}
```
Comment (philloooo) [inline indexd/index/blueprint.py]: need unit tests for new feature

**Comment by philloooo** [comment]
check codacy

**Comment by christiandompierre** [comment]
these codacy checks are stupid. In 2 out of 3 they say the parameters differ but they don't. In the other case, of course they changed, that's the whole point of this PR

**Pull request merged**